proto/p4/config/v1/p4info.proto

```diff
@@ -85,7 +85,8 @@ message P4Ids {
     TABLE = 0x02;
     VALUE_SET = 0x03;
     CONTROLLER_HEADER = 0x04;
-
+    HEADER_OR_STRUCT = 0x05;  // used to test specific struct or header other than
```

Comment:
I don't know what this is for. If this is just for testing your code, it should not be merged into p4info.

Reply:
Sorry, I forgot to add a note that this change could be used for a struct for special handling.  Maybe, even a header that is not used by packet_in/packet_out.  I am using this entry in ```control-plane/p4RuntimeSerializer.cpp``` as follows:

```id = symbols.getId(P4RuntimeSymbolType::HEADER_OR_STRUCT(), name);```

Then the id is used to setup the Preamble.

```setPreamble(header->mutable_preamble(), id, name, name, annotations);```

If an alternative can be suggested how to get the id without using the new HEADER_OR_STRUCT, I can remove this change.

Reply:
You'll have to give a concrete example of when this is required by a runtime application. For now it is completely at odds with the other entities defined in the P4Info message.